src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -0,0 +1,233 @@
+"""
+FEWS NET processing.
+
+Download and save the data provided by FEWS NET.
+"""
+
+import logging
+import zipfile
+from datetime import datetime
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import Union
+
+from requests.exceptions import HTTPError
+
+from aatoolbox.utils.io import download_url, unzip
+
+logger = logging.getLogger(__name__)
+BASE_URL_COUNTRY = (
+    "https://fdw.fews.net/api/ipcpackage/"
+    "?country_code={iso2}&collection_date={YYYY}-{MM}-01"
+)
+BASE_URL_REGION = (
+    "https://fews.net/data_portal_download/download"
+    "?data_file_path=http://shapefiles.fews.net.s3.amazonaws.com/"
+    "HFIC/{region_code}/{region_name}{YYYY}{MM}.zip"
+)
+
+
+def download_zip(
+    url: str,
+    zip_filename: str,
+    output_dir: Path,
+) -> bool:
+    """
+    Download and unzip the file at the url.
+
+    Parameters
+    ----------
+    url : str
+        url that contains the zip file to be downloaded
+    zip_filename : str
+        name of the zipfile
+    output_dir : Path
+        path of dir to which the zip content should be written
+
+    Returns
+    -------
+    valid_file: bool
+        if True, the url contains a valid zip file
+
+    """
+    valid_file = False
+
+    try:
+        # create tempdir to write zipfile to
+        with TemporaryDirectory() as temp_dir:
+            zip_path = Path(temp_dir) / zip_filename
+            download_url(url=url, save_path=zip_path)
+            logger.info(f"Downloaded {url} to {zip_path}")
+
+            try:
+                unzip(zip_file_path=zip_path, save_dir=output_dir)
+                logger.debug(f"Unzipped to {output_dir}")
+                valid_file = True
+            except zipfile.BadZipFile:
+                # indicates that the url returned something that wasn't a
+                # zip, happens often and indicates data for the given
+                # country - date is not available
+                logger.debug(
```

Comment:
Ah sorry, I understand the structure better now. You're right that you can leave it as debug, but then I would do the following:
- make sure you set `valid_file = False` in the `except`
- On the very last line of this module, instead of just logging an error, I would raise an error:
```python     
if not region_data:
    raise RuntimeError(f"No data found for {date.strftime('%Y-%m')}")
```